### PR TITLE
Sprint 3 pull request to master branch 

### PR DESCRIPTION
**Pull request by bbusagroningen**
Sprint 3 includes the following features:
Dgeni documentation 
Create new fields with connections
Change file structure
Test for login
Test for pages
Add titles to pages

### CODE REVIEW

**Comment by killje** [comment]
Is this now done through the right "How to git" method?

File: Switchboard/docs/config/processors/api-data.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                                    // Private isn't set to true, just to an empty string if @private is supplied
+                                    return doc.docType !== 'module';
+                                })
+
+                                .sortBy(function(doc){
+                                    
+                                    var moduleParts = doc.name.split('/');
+                                    moduleParts.pop();
+
+                                    if (moduleParts.length === 0) {
```
Comment (lauraBaakman) [inline Switchboard/docs/config/processors/api-data.js]: Multiple levels of abstraction.

File: Switchboard/docs/config/processors/docLinks.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            };
+
+            parseDocs(docs);
+
+            var parseName = function (name) {
+
+                var targetBlank = true;
+                var href;
+                var type;
+                if (linkUrls[name] !== undefined) {
```
Comment (lauraBaakman) [inline Switchboard/docs/config/processors/docLinks.js]: Multiple levels of abstraction, use a function to tell me what is happening in the body of the if.

File: Switchboard/docs/config/processors/docLinks.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                returnLink += 'href="' + href + '">' + name + '</a></span>';
+                
+                return returnLink;
+                
+            };
+
+            var parseType = function (type) {
+                switch (type.type) {
+                    case 'TypeApplication':
+                        if (type.expression.name === 'Array') {
```
Comment (lauraBaakman) [inline Switchboard/docs/config/processors/docLinks.js]: MLA (Multiple Levels of Abstraction)

File: Switchboard/docs/config/processors/docLinks.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                        }
+                        break;
+                    case 'AllLiteral':
+                        return '<span class="parseType parseType-all">*</span>';
+                        break;
+                    case 'FunctionType':
+                        return parseName('Function');
+                        break;
+                    case 'TypeUnion':
+                        var parsedElements = [];
```
Comment (lauraBaakman) [inline Switchboard/docs/config/processors/docLinks.js]: MLA

File: Switchboard/docs/config/processors/docLinks.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                        _.forEach(type.elements, function (element) {
+                            parsedElements.push(parseType(element));
+                        });
+                        return '(' + parsedElements.join('|') + ')';
+                        break;
+                    default:
+                        var parsedName = '';
```
Comment (lauraBaakman) [inline Switchboard/docs/config/processors/docLinks.js]: MLA

File: Switchboard/public/app/shared/ResourcesFactory.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+     * All information about what urls to use are here
+     * 
+     * @param {ngResource/service/$resource} $resource The angular resource 
+     * service. This is an api to connect to the back-end
+     * 
+     */
+    var ResourcesFactory = function ($resource) {
+
+        // The backend address
+        const BACKEND_RESOURCE = "http://localhost:8484/";
```
Comment (lauraBaakman) [inline Switchboard/public/app/shared/ResourcesFactory.js]: You might want to **consider** pulling stuff like this to a separate configuration file that is read on start up.

Comment (lauraBaakman) [inline Switchboard/public/app/shared/ResourcesFactory.js]: Really only consider it, I am not convinced it is worth the trouble.

File: Switchboard/docs/config/processors/api-data.js (hunk 1 above)
Comment (killje) [inline Switchboard/docs/config/processors/api-data.js]: Is this really needed to edit this for the stuff from the documentation?

Comment (lauraBaakman) [inline Switchboard/docs/config/processors/api-data.js]: It is still code that will rot if you don't take care of it. So yeas, I it really needed.

**Pull request merged**